metricflow/model/validations/agg_time_dimension.py

```diff
@@ -0,0 +1,66 @@
+from typing import List
+
+from metricflow.model.objects.data_source import DataSource
+from metricflow.model.objects.elements.dimension import DimensionType
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.validations.validator_helpers import (
+    ModelValidationRule,
+    ValidationIssueType,
+    validate_safely,
+    ValidationError,
+    ValidationIssue,
+)
+from metricflow.specs import TimeDimensionReference
+
+
+class AggregationTimeDimensionRule(ModelValidationRule):
+    """Checks that the aggregation time dimension for a measure points to a valid time dimension in the data source."""
+
+    @staticmethod
+    @validate_safely(whats_being_done="checking aggregation time dimension for data sources in the model")
+    def validate_model(model: UserConfiguredModel) -> List[ValidationIssueType]:  # noqa: D
+        issues: List[ValidationIssueType] = []
+        for data_source in model.data_sources:
+            issues.extend(AggregationTimeDimensionRule._validate_data_source(data_source))
+
+        return issues
+
+    @staticmethod
+    def _time_dimension_in_model(time_dimension_reference: TimeDimensionReference, data_source: DataSource) -> bool:
+        for dimension in data_source.dimensions:
+            if dimension.type == DimensionType.TIME and dimension.name == time_dimension_reference.element_name:
+                return True
+        return False
+
+    @staticmethod
+    @validate_safely(whats_being_done="checking aggregation time dimension for a data source")
+    def _validate_data_source(data_source: DataSource) -> List[ValidationIssueType]:
+        issues: List[ValidationIssueType] = []
+
+        for measure in data_source.measures:
+            model_object_reference = ValidationIssue.make_object_reference(
+                data_source_name=data_source.name, measure_name=measure.name
+            )
+            if not measure.agg_time_dimension:
+                issues.append(
+                    ValidationError(
+                        model_object_reference=model_object_reference,
+                        message=f"In data source '{data_source.name}', measure '{measure.name}' does not have an "
+                        f"aggregation time dimension set",
+                    )
+                )
+                continue
+            agg_time_dimension_reference = measure.checked_agg_time_dimension
```

Comment:
This throws the assertion we need. Can we use that here instead of converting it to a validation error as above? 

I don't think we ever want a code path where someone runs the validators on an un-transformed model, because it'll lead to potentially huge numbers of validation failures. Seems better to indicate the order of operations flaw at the earliest possible moment in cases, like this one, where the only reason this value isn't set is because the model hasn't been processed correctly.

Reply:
Updated.